test/exe/main_common_test.cc

```diff
@@ -310,7 +310,7 @@ TEST_P(MainCommonTest, ConstructDestructLogger) {
   VERBOSE_EXPECT_NO_THROW(MainCommon main_common(argc(), argv()));
 
   const std::string logger_name = "logger";
-  spdlog::details::log_msg log_msg;
+  static spdlog::details::log_msg log_msg;
```

Comment:
This seems like a bug in the upstream library, right? Just checking my understanding. It might be worth a comment here and an upstream issue as this is kind of strange?

Reply:
TL;DR: Will do, but extra context below.

So I'm 80% sure it's a bug in the upstream library :-}.  The things that go into that 20% are:
* I can't find documentation on how spdlog is supposed to be used.
* spdlog::details::log_msg looks like the kind of structure that consumers shouldn't be using directly.
* This is the only place in envoy that I could find that references either that structure or Logger::Registry::getSink(), and neither of those things strike me as having much to do with what the test says it's testing.
But this is all in the context of a lack of expertise on my part around the use of spdlog by envoy.

However, I think it's likely in the extreme that this is not the ideal fix; that fix would be one or both of rewriting this test to actually use the same APIs for logging that the rest of envoy does or fixing things upstream.  I just wasn't sure how much energy should be put into an uninitialized memory reference from a test.  

Let me know if given that context you want something more or different from an upstream bug and a reference to it from here.

Reply:
It's not well-named, but I think this test is checking that the logger is still functional after MainCommon is destroyed. Presumably this is to ensure that final log messages stand a chance of making it out to the world.

It was crashing after the upgrade to the new version of spdlog because of a null logger name  (formerly we just called `Logger::Registry::getSink()->log({})` where `{}` is a spdlog::details::log_msg). I didn't look at log_msg closely enough to notice that it's default constructor doesn't initialize all the struct's fields. There's another constructor (which takes logger name and level) which seems to be used in spdlog code. Perhaps that's a better choice for this test?

Reply:
If we leave this code as written can we have comments explaining why it's static?

I'm the one that originally wrote:   `Logger::Registry::getSink()->log({});` could we instead replace that with an ENVOY_LOG call?